internal/generate/clusterserviceversion/clusterserviceversion.go

```diff
@@ -226,12 +264,47 @@ func (g Generator) makeKustomizeBaseGetter(inputDir, apisDir string, ilvl projut
 // makeBaseGetter returns a function that gets a base from inputDir.
 // apisDir is used by getBaseFunc to populate base fields.
 func (g Generator) makeBaseGetter(basePath, apisDir string, interactive bool) getBaseFunc {
-	gvks := make([]schema.GroupVersionKind, len(g.config.Resources))
-	for i, gvk := range g.config.Resources {
-		gvks[i].Group = fmt.Sprintf("%s.%s", gvk.Group, g.config.Domain)
-		gvks[i].Version = gvk.Version
-		gvks[i].Kind = gvk.Kind
+	var gvks []schema.GroupVersionKind
+	if g.Collector != nil {
+		if g.Collector.V1CustomResourceDefinitions != nil {
+			for _, crd := range g.Collector.V1CustomResourceDefinitions {
+				for _, version := range crd.Spec.Versions {
+					gvks = append(gvks, schema.GroupVersionKind{
+						Group:   crd.Spec.Group,
+						Version: version.Name,
+						Kind:    crd.Spec.Names.Kind,
+					})
+				}
+			}
+		}
+		if g.Collector.V1beta1CustomResourceDefinitions != nil {
+			for _, crd := range g.Collector.V1beta1CustomResourceDefinitions {
+				for _, version := range crd.Spec.Versions {
+					gvks = append(gvks, schema.GroupVersionKind{
+						Group:   crd.Spec.Group,
+						Version: version.Name,
+						Kind:    crd.Spec.Names.Kind,
+					})
+				}
+			}
+		}
+	} else if g.config != nil {
+		fmt.Printf("\ng.Collector is nil\nbasepath:\t%v\napisdir:\t%v\n", basePath, apisDir)
+
+		gvks = make([]schema.GroupVersionKind, len(g.config.Resources))
+		for i, gvk := range g.config.Resources {
+			/*gvks = append(gvks, schema.GroupVersionKind{
+				Group: fmt.Sprintf("%s.%s", gvk.Group, g.config.Domain),
+				Version: gvk.Version,
+				Kind: gvk.Kind,
+			})*/
+			gvks[i].Group = fmt.Sprintf("%s.%s", gvk.Group, g.config.Domain)
+			gvks[i].Version = gvk.Version
+			gvks[i].Kind = gvk.Kind
+			fmt.Printf("\n\nGVK's in cfg: \n\n%+v\n\n", gvk)
+		}
```

Comment:
You should always be able to use CRDs in the collector.
```suggestion
```

Reply:
I take this back, I don't think this is possible with `generate kustomize manifests`. How about creating `makeBaseGetter()` and `makeBaseCreator()`, the former using CRDs and the latter using the config?

Reply:
That would work and it would be cleaner.